chainer/training/extensions/computational_graph.py

```diff
@@ -60,37 +60,45 @@ def dump_graph(root_name, out_name='cg.dot',
        for the ``variable_style`` and ``function_style`` arguments.
 
     """
-    def trigger(trainer):
-        return trainer.updater.iteration == 1
-
-    if variable_style is None:
-        variable_style = _var_style
-    if function_style is None:
-        function_style = _func_style
-
-    original_flag = [None]
-
-    def initializer(_):
-        original_flag[0] = configuration.config.keep_graph_on_report
-        configuration.config.keep_graph_on_report = True
 
-    @extension.make_extension(trigger=trigger, initializer=initializer)
-    def dump_graph(trainer):
+    def __init__(self, root_name, out_name='cg.dot',
+                 variable_style=None, function_style=None):
+        self._root_name = root_name
+        self._out_name = out_name
+        self._variable_style = variable_style or _var_style
+        self._function_style = function_style or _func_style
+        self._original_flag = None
+        self._flag_called = False
+
+    def initialize(self, _):
```

Comment:
Could you tell me why `_` exists here?

Reply:
The methods are called by https://github.com/chainer/chainer/blob/master/chainer/training/trainer.py#L304 with one argument.
Do you recommend fix the name to `trainer`?

Reply:
Thank you, I just didn't know how `initialize` is invoked.

> Do you recommend fix the name to trainer?
I think it's up to core members.

Reply:
It's safer to use `trainer` as the argument name here because, in Python, the argument name is a part of API (due to keyword argument call).

Reply:
OK, I reflected it.